### PR TITLE
feat: deployment scripts

### PR DESCRIPTION
**Pull request by mattstam**
fully forge script deployment process, and v1.0.8-testnet deployment

uses template changes found in https://github.com/succinctlabs/sp1/pull/963

### CODE REVIEW

File: contracts/script/utils/Base.s.sol [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    /// otherwise just simulate the transaction execution.
+    modifier broadcaster() {
+        vm.startBroadcast(msg.sender);
+        _;
+        vm.stopBroadcast();
+    }
+
+    /// @notice When used, runs the script on the chains specified in the `CHAIN_IDS` env variable.
+    /// Must have a `RPC_${CHAIN_ID}` env variable set for each chain.
+    modifier multichain() {
```
Comment (mattstam) [inline contracts/script/utils/Base.s.sol:21]: @ctian1

File: contracts/script/deploy/SP1Verifier.s.sol [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        // Read config
+        bytes32 CREATE2_SALT = readBytes32("CREATE2_SALT");
+        address SP1_VERIFIER_GATEWAY = readAddress("SP1_VERIFIER_GATEWAY");
+
+        // Deploy contract
+        address verifier = address(new SP1Verifier{salt: CREATE2_SALT}());
+
+        // Add the verifier to the gateway
+        SP1VerifierGateway gateway = SP1VerifierGateway(SP1_VERIFIER_GATEWAY);
+        gateway.addRoute(verifier);
```
[the remark this answers is not in the document]
Comment (mattstam) [reply contracts/script/deploy/SP1Verifier.s.sol]: wdym? Each verifier version corresponds to one route

File: README.md [hunk 3]
```diff
@@ -14,6 +15,7 @@ forge install succinctlabs/sp1-contracts
 ```
 
 To install a specific version:
+
 ```bash
 forge install succinctlabs/sp1-contracts@<version>
```
[the remark this answers is not in the document]
Comment (mattstam) [reply README.md:20]: I think it's still relevant? If someone wants these contracts as a dependency, they can specify a specific tag/commit this way

File: contracts/script/deploy/SP1VerifierGateway.sol [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import {BaseScript} from "../utils/Base.s.sol";
+import {SP1VerifierGateway} from "../../src/SP1VerifierGateway.sol";
+
+contract SP1VerifierGatewayScript is BaseScript {
+    string internal constant KEY = "SP1_VERIFIER_GATEWAY";
+
+    function run() external multichain(KEY) broadcaster {
+        // Read config
+        bytes32 CREATE2_SALT = readBytes32("CREATE2_SALT");
+        address OWNER = readAddress("OWNER");
```
[the remark this answers is not in the document]
Comment (mattstam) [reply contracts/script/deploy/SP1VerifierGateway.sol:14]: updated

[the remark this answers is not in the document]
Comment (mattstam) [reply contracts/script/deploy/SP1VerifierGateway.sol]: updated

File: contracts/.env.example [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ETHERSCAN_API_URL_11155111=https://api-sepolia.etherscan.io/api
+ETHERSCAN_API_URL_100=https://api.gnosisscan.io/api
+ETHERSCAN_API_URL_137=https://api.polygonscan.com/api
+ETHERSCAN_API_URL_420=https://api-optimistic.etherscan.io/api
+ETHERSCAN_API_URL_42161=https://api.arbiscan.io/api
+ETHERSCAN_API_URL_421614=https://api-sepolia.arbiscan.io/api
+ETHERSCAN_API_URL_8453=https://api.basescan.org/api
+ETHERSCAN_API_URL_84532=https://api-sepolia.basescan.org/api
+ETHERSCAN_API_URL_534352=https://api.scrollscan.com/api
+ETHERSCAN_API_URL_534351=https://api-sepolia.scrollscan.com/api
```
[the remark this answers is not in the document]
Comment (mattstam) [reply contracts/.env.example]: updated to chain names

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ctian1) [reply README.md:20]: but shouldn't you only need latest?

File: contracts/foundry.toml [hunk 6]
```diff
@@ -3,11 +3,41 @@ src = "src"
...
 line_length = 100
 tab_width = 4
 func_attrs_with_params_multiline = true
 ignore = ["lib/**"]
 
-# See more config options https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
\ No newline at end of file
+# See more config options https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
+
+[rpc_endpoints]
```
Comment (ctian1) [inline contracts/foundry.toml:23]: IMO we should just not need these if CHAINS is user specified

**Review by ctian1 — APPROVED** [review]

File: contracts/foundry.toml (hunk 6 above)
Comment (mattstam) [reply contracts/foundry.toml:23]: I think its nice to have some examples of how it all connects together. Plus having them all here makes it so if anyone else needs to deploy the contracts, they can git clone and go without having to configure stuff

**Pull request merged**